fsharp-backend/src/LibExecution/StdLib/LibString.fs

```diff
@@ -749,81 +749,81 @@ Don't rely on either the size or the algorithm."
-//       "Parse a UUID of form XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX from the input `uuid` string"
-//      ; fn =
-//         (function
-//         | _, [DStr s] ->
-//           ( match Uuidm.of_string (Unicode_string.to_string s) with
-//           | Some id ->
-//               DUuid id
-//           | None ->
-//               Exception.code
-//                 "`uuid` parameter was not of form XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
-//           )
-//         | args ->
-//             incorrectArgs ())
-//      ; sqlSpec = NotYetImplementedTODO
-//      ; previewable = Pure
-//      ; deprecated = ReplacedBy(fn "" "" 0) }
-//      { name = fn "String" "toUUID" 1
-//      ; parameters = [Param.make "uuid" TStr]
-//      ; returnType = TResult
-//      ; description =
-//       "Parse a UUID of form XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX from the input `uuid` string"
-//      ; fn =
-//         (function
-//         | _, [DStr s] ->
-//           ( match Uuidm.of_string (Unicode_string.to_string s) with
-//           | Some id ->
-//               DResult (ResOk (DUuid id))
-//           | None ->
-//               error_result
-//                 "`uuid` parameter was not of form XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
-//           )
-//         | args ->
-//             incorrectArgs ())
-//      ; sqlSpec = NotYetImplementedTODO
-//      ; previewable = Pure
-//      ; deprecated = NotDeprecated }
-//      { name = fn "String" "isSubstring" 0
-//      ; parameters = [Param.make "searchingFor" TStr; Param.make "lookingIn" TStr]
-//      ; returnType = TBool
-//      ; description = "Checks if `lookingIn` contains `searchingFor`"
-//      ; fn =
-//         (function
-//         | _, [DStr needle; DStr haystack] ->
-//             DBool (Unicode_string.is_substring needle haystack)
-//         | args ->
-//             incorrectArgs ())
-//      ; sqlSpec = NotYetImplementedTODO
-//      ; previewable = Pure
-//      ; deprecated = ReplacedBy(fn "" "" 0) }
-//      { name = fn "String" "isSubstring" 1
-//      ; parameters = [Param.make "lookingIn" TStr; Param.make "searchingFor" TStr]
-//      ; returnType = TBool
-//      ; description = "Checks if `lookingIn` contains `searchingFor`"
-//      ; fn =
-//         (function
-//         | _, [DStr haystack; DStr needle] ->
-//             DBool (Unicode_string.is_substring needle haystack)
-//         | args ->
-//             incorrectArgs ())
-//      ; sqlSpec = NotYetImplementedTODO
-//      ; previewable = Pure
-//      ; deprecated = ReplacedBy(fn "" "" 0) }
-    { name = fn "String" "contains" 0
+    { name = fn "String" "toUUID" 0
+      parameters = [ Param.make "uuid" TStr "  " ]
+      returnType = TUuid
+      description =
+        "Parse a UUID of form XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX from the input `uuid` string"
+      fn =
+        (function
+        | _, [ DStr s ] ->
+            match s with
+            | Parse (x : System.Guid) -> System.Guid.Parse s |> DUuid |> Value
```

Comment:
I dont understand what's happening here? How does a String match with `| Parse`? Is this an active pattern?

Reply:
Yes, `Parse` is an old active pattern defined in F#+ which use a generic `tryParse` function internally. The name is a bit unfortunate, as it should be called `Parsed` instead.

So, doing `| Parse (x : System.Guid) -> System.Guid.Parse s |` makes no sense as `x` contains the already "parsed" value, so here we're parsing twice.

Also, since `Parse` is an active patters it could be combined with the DU pattern in one, so:

```
(function
    | _, [DStr (Parse x)] -> x |> DUuid |> Value
    | _ -> ....
```
and as you can see the type annotation is not needed, since there is a `DUuid` which AFAIK it accepts a GUID which makes enough context for type inference.

Hope this helps !
should do the work.

Reply:
Thank you @gusty for the great explanation :)

Reply:
I'd prefer to avoid using f#+ where possible (esp in stdlib). I'm also a little weary of active patterns as they obscure what's happening a bit imo. Would you mind doing the tryParse directly?

Reply:
Understood @pbiggar, now I'm aware of that  :) I pushed the changes. Thank you.